simply/market.py

```diff
@@ -24,9 +24,8 @@ def __init__(self, time, network=None, grid_fee_matrix=None):
         self.network = network
         self.save_csv = cfg.parser.getboolean("default", "save_csv", fallback=False)
         self.csv_path = Path(cfg.parser.get("default", "path", fallback="./scenarios/default"))
-        if grid_fee_matrix is not None:
-            self.grid_fee_matrix = grid_fee_matrix
-        elif network is not None:
+        self.grid_fee_matrix = grid_fee_matrix
+        if network is not None:
```

Comment:
My intention was to move the assignment of the grid_fee_matrix instance variable outside of an 'if' statement so it can be used to determine whether to apply the grid fee (e.g. line 156 - if self.grid_fee_matrix == None then don't apply grid fee). I have now updated the assignment so it prioritises a grid_fee_matrix assigned explicitly over a grid_fee_matrix from the 'network' parameter. I have also added a test that checks this is the case.